### DISCUSSION TITLE
Post Build action on failed build

### DISCUSSION THREAD

**Discussion by caleb654** [codemagic-ci-cd/codemagic-docs#2300] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Is there any way to run a command in the post build step only if the build fails? I am using the work flow editor.

### Expected results

N/A

### Actual results

N/A

**Comment by dtrdic** [comment, marked as answer]
hey @caleb654, yes you can do it. You can write a condition in the post-publish step like this

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/b1cd2467-a86b-4e88-833d-f90bc2bf53a4)

**Comment by dtrdic** [comment]
I am closing the thread due to inactivity. Let me know if you have any more questions by replying to this thread or opening a new one.
